cdap-ui/app/cdap/components/AbstractWidget/DLPCustomWidget/DLPRow.tsx

```diff
@@ -0,0 +1,449 @@
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import FormControl from '@material-ui/core/FormControl';
+import ConfigurationGroup, { IConfigurationGroupProps } from 'components/ConfigurationGroup';
+import { IErrorObj } from 'components/ConfigurationGroup/utilities';
+
+const styles = (theme): StyleRules => {
+  return {
+    ...AbstractRowStyles(theme),
+    root: {
+      minHeight: '50px',
+      display: 'grid',
+      gridTemplateColumns: '1fr auto auto auto',
+      alignItems: 'center',
+      '& > button': {
+        width: '40px',
+        height: '40px',
+        margin: 'auto',
+        marginTop: '5px',
+      },
+    },
+    inputContainer: {
+      display: 'grid',
+      gridTemplateColumns: '40px 1fr 30px 1fr 50px 1fr',
+      gridGap: '10px',
+    },
+    errorBorder: {
+      border: '2px solid',
+      borderColor: theme.palette.red[50],
+    },
+    disabled: {
+      color: `${theme.palette.grey['50']}`,
+    },
+    separator: {
+      textAlign: 'center',
+    },
+    group: {
+      marginLeft: '20px',
+      marginTop: '15px',
+      marginBottom: '15px',
+      '& > div > div': {
+        padding: '15px 10px 10px',
+      },
+    },
+    groupTitle: {
+      marginBottom: '15px',
+      '& > h2': {
+        fontSize: '1.2rem',
+      },
+    },
+    errorText: {
+      color: theme.palette.red[50],
+      marginLeft: '15px',
+      marginTop: '5px',
+    },
+  };
+};
+
+interface IErrorConfig {
+  transform: string;
+  fields: string;
+  filters: string;
+  transformPropertyId: string;
+  isNestedError: boolean;
+}
+
+export interface ITransformProp {
+  label: string;
+  name: string;
+  options: IWidgetProperty[];
+  supportedTypes?: string[];
+  filters?: IPropertyFilter[];
+}
+export type FilterOption = IOption | string;
+
+interface IDLPRowProps extends IAbstractRowProps<typeof styles> {
+  transforms: ITransformProp[];
+  filters: FilterOption[];
+  extraConfig: any;
+}
+
+interface IDLPRowState {
+  fields: string;
+  transform: string;
+  filters: string;
+  transformProperties: Record<string, string>;
+  expanded?: boolean;
+}
+
+type StateKeys = keyof IDLPRowState;
+
+class DLPRow extends AbstractRow<IDLPRowProps, IDLPRowState> {
+  public static defaultProps = {
+    transforms: [],
+    filters: [],
+    extraConfig: '',
+    expanded: false,
+  };
+
+  public state = {
+    fields: '',
+    transform: '',
+    filters: '',
+    transformProperties: {},
+    expanded: false,
+  };
+
+  public componentDidMount() {
+    try {
+      const jsonString = this.props.value;
+      const jsonObj = JSON.parse(jsonString);
+      this.setState({ ...this.state, ...jsonObj });
+    } catch (error) {
+      this.setState({
+        fields: '',
+        transform: '',
+        filters: '',
+        transformProperties: {},
+      });
+    }
+  }
+
+  // Handles onChange for MultiSelect components (filters and fields selectors)
+  private handleChangeMultiSelect = (type: StateKeys, e: string) => {
```

Comment:
since these handling of the individual inputs seemed non-trivial, might be better to extract them each to their own component and handle the change transformation there

Reply:
It will be non-trivial to put these into their own components because one Select change handler changes the state for another value (ex. the Transform select changes the contents of the transformProperties object in state). 

I could separate this handler into two handlers, each with a more detailed purpose if that would help readability?